### DISCUSSION TITLE
Creating a new s3 cache bucket

### DISCUSSION THREAD

**Discussion by lmilbaum** [npalm/terraform-aws-gitlab-runner#966] [Q&A]
I have one environment deployed in one region using s3 cache bucket. Now I want to deploy another environment in a second region with a different s3 cache bucket. 
I'm using terragrunt to keep different state files for each environment. For unknown reason when running the plan for the second environment, it picks up the first environment s3 cache bucket.

How do I go about it?

**Comment by lmilbaum** [comment]
I've tried adding the following attribute `runner_worker_cache_s3_bucket_prefix` which thrown the following error:

```
 prefix=[/workspaces/gitlab-runners/live/dev/runner] 
INFO[0010] ╷
│ Error: Unsupported argument
│ 
│   on main.tf line 127, in module "runner":
│  127:   runner_worker_cache_s3_bucket_prefix = var.environment
│ 
│ An argument named "runner_worker_cache_s3_bucket_prefix" is not expected
│ here.
```

**Comment by lmilbaum** [comment]
Is it possible to apply the module multiple times in the same AWS account even if using different state files?

**Reply by kayman-mk** [reply]
Should work as long as you use a different `environment` for every module. Otherwise the resource names will clash in your AWS account.

**Reply by lmilbaum** [reply]
IC. Thanks for your answer. It somehow clashes with the terragrunt environments.
